### PR TITLE
Start adding FreeBSD support to scripts

### PR DESCRIPTION
**Pull request by freqlabs**
* Convert critical /bin/bash scripts to /bin/sh
* Add FreeBSD*) cases to some critical scripts
* Update master.cfg to add some FreeBSD workers

### CODE REVIEW

File: scripts/bb-build.sh [hunk 1]
```diff
@@ -87,13 +98,18 @@ packages|kmod|pkg-kmod|dkms|dkms-kmod)
 	;;
 in-tree)
 	./configure $CONFIG_OPTIONS $LINUX_OPTIONS >>$CONFIG_LOG 2>&1 || exit 1
-	make $MAKE_OPTIONS >>$MAKE_LOG 2>&1 || exit 1
+	$MAKE $MAKE_OPTIONS >>$MAKE_LOG 2>&1 || exit 1
 	./scripts/zfs-tests.sh -cv >>$INSTALL_LOG 2>&1
 	sudo -E scripts/zfs-helpers.sh -iv >>$INSTALL_LOG 2>&1
 	;;
+freebsd)
```
Comment (behlendorf) [inline scripts/bb-build.sh]: Nothing here looks to be platform specific, why don't we make this a generic install method called "system".

**Comment by freqlabs** [comment]
Rebased.
In my test env the FreeBSD VM user-data script shows up in the text console log now and the bootstrap script makes it to starting the slave, but I haven't set up a master to see how it goes after that.

**Comment by behlendorf** [comment]
This is looking good.  I was able to test the updated PR and everything is nicely bootstrapping now.

**Comment by behlendorf** [comment]
This functionality was merged.

**Pull request closed without merging**